mypy/typeanal.py

```diff
@@ -824,6 +805,35 @@ def anal_type(self, tp: UnboundType) -> Type:
 TypeVarList = List[Tuple[str, TypeVarExpr]]
 
 
+def fix_instance(t: Instance, fail: Callable[[str, Context], None]) -> None:
+    """Fix a malformed instance by replacing all type arguments with Any.
+
+    Also emit a suitable error if this is not due to implicit Any's.
+    """
+    if len(t.args) == 0:
+        any_type = AnyType(TypeOfAny.from_omitted_generics,
+                           line=t.line, column=t.column)
+        t.args = [any_type] * len(t.type.type_vars)
+        return
+    # Invalid number of type parameters.
+    n = len(t.type.type_vars)
+    s = '{} type arguments'.format(n)
+    if n == 0:
+        s = 'no type arguments'
+    elif n == 1:
+        s = '1 type argument'
+    act = str(len(t.args))
+    if act == '0':
+        act = 'none'
+    fail('"{}" expects {}, but {} given'.format(
+        t.type.name(), s, act), t)
+    # Construct the correct number of type arguments, as
+    # otherwise the type checker may crash as it expects
+    # things to be right.
+    t.args = [AnyType(TypeOfAny.from_error) for _ in t.type.type_vars]
+    t.invalid = True
```

Comment:
Did anything change here, or is it a pure refactor?

Reply:
This is a pure refactor. I first tried a different solution, that required calling this manually from other module, but then found a simpler solution. However, looking at it, it seemed to me it actually looks better as a separate function (all visitor methods look compact this way).

Reply:
Agreed, just wanted to check I wasn't missing anything